### PR TITLE
csgrep --set-scan-prop: override the specified scan property

### PR DESCRIPTION
**Pull request by kdudka**
Fixes: https://github.com/csutils/csdiff/issues/82

### CODE REVIEW

**Comment by jamacku** [comment]
Thank you so much for working on this, I'm going to test it in `differential-shellcheck`.

File: src/csgrep.cc [hunk 1]
```diff
@@ -557,17 +600,46 @@ void printUsage(TStream &str, const TDesc &desc)
...
+
 template <class TDecorator>
-bool chainDecorator(
+bool chainDecoratorIntArg(
         AbstractWriter            **pEng,
         const po::variables_map    &vm,
         const char                 *key)
 {
-    if (!vm.count(key))
+    const po::variables_map::const_iterator it = vm.find(key);
```
Comment (lzaoral) [inline src/csgrep.cc:637]:
```suggestion
    const auto it = vm.find(key);
```

**Comment by kdudka** [comment]
@lzaoral Thanks for review!  I have updated the pull request.

**Review by lzaoral — APPROVED** [review]
LGTM!  Thanks!  Let's wait for @jamacku's review before merging though.

**Review by jamacku — APPROVED** [review]
@kdudka @lzaoral, Tested in:

- https://github.com/redhat-plumbers-in-action/differential-shellcheck/pull/124

Works great, thanks!

**Comment by kdudka** [comment]
@jamacku Perfect, thank you for testing it!

**Pull request merged**